relay-common/src/alloc.rs

```diff
@@ -0,0 +1,40 @@
+use std::alloc::{GlobalAlloc, Layout, System};
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+struct RelayAllocator {
+    max_size: AtomicUsize,
+}
+
+unsafe impl GlobalAlloc for RelayAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let max_size = self.max_size.load(Ordering::Relaxed);
```

Comment:
What is the overhead of reading an atomic variable on every allocation?

Reply:
I believe it should be negligible with `Ordering::Relaxed`

Reply:
I also wouldn't worry about performance here.
In the grand scheme of things, the time for getting the value of an AtomicUsize 
(i think) is dwarfed by the time for the allocation function itself.